Sources/NIO/SocketChannel.swift

```diff
@@ -489,36 +500,94 @@ final class DatagramChannel: BaseSocketChannel<Socket> {
         var rawAddressLength = socklen_t(MemoryLayout<sockaddr_storage>.size)
         var buffer = self.recvAllocator.buffer(allocator: self.allocator)
         var readResult = ReadResult.none
-
-        for i in 1...self.maxMessagesPerRead {
-            guard self.isOpen else {
-                throw ChannelError.eof
-            }
-            buffer.clear()
-
-            let result = try buffer.withMutableWritePointer {
-                try self.socket.recvfrom(pointer: $0, storage: &rawAddress, storageLen: &rawAddressLength)
+        
+        return try self.selectableEventLoop.withControlMessageBytes {
+            let controlBytes = self.reportExplicitCongestionNotifications ?
+                    $0:
+                    UnsafeMutableRawBufferPointer(start: nil, count: 0)
```

Comment:
I think if you're splitting the ternary over three lines it's not really aiding clarity anymore. Better to replace with `if`/`else` I think.